pkg/operator/ceph/cluster/osd/osd.go

```diff
@@ -263,13 +264,41 @@ func (c *Cluster) startProvisioningOverPVCs(config *provisionConfig) {
 			continue
 		}
 
+		//Skip OSD prepare if deployement already exists for the PVC
+		listOpts := metav1.ListOptions{LabelSelector: fmt.Sprintf("%s=%s,%s=%s",
+			k8sutil.AppAttr, AppName,
+			pvcLabelKey, osdProps.crushHostname,
```

Comment:
I see the crushHostName set above as the PVC name above on line 253, but I'd suggest we set this to `volume.PersistentVolumeClaimSource.ClaimName` just to be clear that we want the pvc name and not the host name, just for code readability. The crushHostname is actually not the pvc name if the osd is not portable, this must be updated somewhere else.